aten/src/ATen/native/RangeFactories.cpp

```diff
@@ -141,11 +141,18 @@ Tensor& arange_cpu_out(Tensor& result, Scalar start, Scalar end, Scalar step) {
 
     TORCH_CHECK(size_d >= 0 && size_d <= static_cast<double>(std::numeric_limits<int64_t>::max()),
              "invalid size, possible overflow?");
+
     int64_t size = static_cast<int64_t>(size_d);
+    int64_t numel = result.numel();
 
-    if (result.numel() != size) {
+    if (numel != size) {
+      if(numel > 0){
```

Comment:
This message should be improved.  For example, it never says what the computed result size is.

How about something like:
"The number of elements in the out tensor ", result.numel(), " of shape ", out.sizes(), " does not match the computed number of elements ", size, ".  Note that this may occur as a result of rounding error.  The out tensor will be resized to a tensor of shape "(", size, ",)."